js/ui/date_range_box.d.ts

```diff
@@ -0,0 +1,119 @@
+import {
+  EventInfo,
+  NativeEventInfo,
+  InitializedEventInfo,
+  ChangedOptionInfo,
+} from '../events/index';
+
+import {
+  DropDownButtonTemplateDataModel,
+} from './drop_down_editor/ui.drop_down_editor';
+
+import dxDateBox, { dxDateBoxOptions } from './date_box';
+
+import {
+  ValueChangedInfo,
+} from './editor/editor';
+
+/** @public */
+export type DateRangePickerType = 'calendar' | 'native';
+
+/** @public */
+export type ChangeEvent = NativeEventInfo<dxDateRangeBox>;
+
+/** @public */
+export type ClosedEvent = EventInfo<dxDateRangeBox>;
+
+/** @public */
+export type ContentReadyEvent = EventInfo<dxDateRangeBox>;
+
+/** @public */
+export type CopyEvent = NativeEventInfo<dxDateRangeBox, ClipboardEvent>;
+
+/** @public */
+export type CutEvent = NativeEventInfo<dxDateRangeBox, ClipboardEvent>;
+
+/** @public */
+export type DisposingEvent = EventInfo<dxDateRangeBox>;
+
+/** @public */
+export type EnterKeyEvent = NativeEventInfo<dxDateRangeBox, KeyboardEvent>;
+
+/** @public */
+export type FocusInEvent = NativeEventInfo<dxDateRangeBox, FocusEvent>;
+
+/** @public */
+export type FocusOutEvent = NativeEventInfo<dxDateRangeBox, FocusEvent>;
+
+/** @public */
+export type InitializedEvent = InitializedEventInfo<dxDateRangeBox>;
+
+/** @public */
+export type InputEvent = NativeEventInfo<dxDateRangeBox, UIEvent & { target: HTMLInputElement }>;
+
+/** @public */
+export type KeyDownEvent = NativeEventInfo<dxDateRangeBox, KeyboardEvent>;
+
+/** @public */
+export type KeyPressEvent = NativeEventInfo<dxDateRangeBox, KeyboardEvent>;
+
+/** @public */
+export type KeyUpEvent = NativeEventInfo<dxDateRangeBox, KeyboardEvent>;
+
+/** @public */
+export type OpenedEvent = EventInfo<dxDateRangeBox>;
+
+/** @public */
+export type OptionChangedEvent = EventInfo<dxDateRangeBox> & ChangedOptionInfo;
+
+/** @public */
+export type PasteEvent = NativeEventInfo<dxDateRangeBox, ClipboardEvent>;
+
+/** @public */
+export type ValueChangedEvent =
+    NativeEventInfo<dxDateRangeBox, KeyboardEvent | MouseEvent | PointerEvent | Event>
+    & ValueChangedInfo;
+
+/** @public */
+export type DropDownButtonTemplateData = DropDownButtonTemplateDataModel;
+
+/**
+ * @public
+ */
+export type Properties = Omit<dxDateBoxOptions<dxDateRangeBox>, 'value' | 'adaptivityEnabled' | 'interval' | 'type' | 'showAnalogClock'> & {
+  /**
+     * @docid dxDateRangeBoxOptions.endDate
+     * @default null
+     * @public
+     */
+  endDate?: Date | number | string;
+  /**
+     * @docid dxDateRangeBoxOptions.pickerType
+     * @default 'calendar'
+     * @default 'native' &for(iOS)
+     * @default 'native' &for(Android)
+     * @public
+     */
+  pickerType?: DateRangePickerType;
```

Comment:
Why we omit `value` from `dxDateBoxOptions<dxDateRangeBox>`, but don't omit a `pickerType`?

Reply:
As I understand, we have no need to omit `pickerType` because its includes types from parent. For example if parent has type `'a' | 'b'`, child can have type `'a'` without omiting. But value has completely enother type

May be ApiReviewers will correct my answer

Reply:
then y do we need to omit `value` and other props?)
@IlyaKhD consult us please

Reply:
1) If you don't omit `value`, it will have merged type:
```ts
((string | number | Date) & (string | number | Date)[]) | undefined
```

instead of
```ts
 (string | number | Date)[] | undefined
```

This happens because `dxDateBoxOptions` and anonymous type are combined with intersection. If a property is presented  in both types, its type will be a union of both possible types.

Reply:
2. AFAIU `'adaptivityEnabled' | 'interval' | 'type' | 'showAnalogClock'>` are omitted because they are just not needed in the new type (`Properties`)

Reply:
3. Technically, we don't have to include `pickerType` in the omit statement, but I would do it to explicitly point to the fact that this prop is overridden.

Why it works at the moment - because when these two types are merged, the result looks like the narrower one:

```ts
type DatePickerType = "calendar" | "list" | "native" | "rollers"
type DateRangePickerType = "calendar" | "native"

type merged = DatePickerType | DateRangePickerType // equals "calendar" | "native"
```